### PR TITLE
Change console.log to deprecationWarning

### PR DESCRIPTION
**Pull request by JudyWeng**
#4313

### CODE REVIEW

File: Source/Scene/Batched3DModel3DTileContent.js [hunk 1]
```diff
@@ -42,6 +44,7 @@ define([
      * {@link https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/TileFormats/Batched3DModel/README.md|Batched 3D Model}
      * tile in a {@link https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/README.md|3D Tiles} tileset.
      *
+     *
```
Comment (lilleyse) [inline Source/Scene/Batched3DModel3DTileContent.js]: Remove this whitespace.

File: Source/Scene/Batched3DModel3DTileContent.js [hunk 2]
```diff
@@ -214,11 +217,12 @@ define([
         // If the header is in the legacy format 'batchLength' will be the start of the JSON string (a quotation mark) or the glTF magic.
         // Accordingly the first byte of uint32 will be either 0x22 or 0x67 and so the uint32 will exceed any reasonable 'batchLength'.
         if (batchLength > 10000000) {
+            //@deprecated b3dm legacy header format [batchLength] [batchTableByteLength]
```
Comment (lilleyse) [inline Source/Scene/Batched3DModel3DTileContent.js]: Remove this.

File: Source/Scene/Batched3DModel3DTileContent.js [hunk 3]
```diff
@@ -214,11 +217,12 @@ define([
...
             byteOffset -= sizeOfUint32;
             batchLength = batchTableJsonByteLength;
             batchTableJsonByteLength = batchTableBinaryByteLength;
             batchTableBinaryByteLength = 0;
-            console.log('Warning: b3dm header is using the legacy format [batchLength] [batchTableByteLength]. The new format is [batchTableJsonByteLength] [batchTableBinaryByteLength] [batchLength] from https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/TileFormats/Batched3DModel/README.md.');
+            deprecationWarning('Batched3DModel3DTileContent.prototype.initialize', 'This b3dm header is using the legacy format [batchLength] [batchTableByteLength]. The new format is [batchTableJsonByteLength] [batchTableBinaryByteLength] [batchLength] from https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/TileFormats/Batched3DModel/README.md.');
```
Comment (lilleyse) [inline Source/Scene/Batched3DModel3DTileContent.js]: Perhaps give a different id just for clarity, something like 'b3dm-legacy-header'

**Comment by lilleyse** [comment]
Just some small comments, I tested it and it works well!

**Comment by lilleyse** [comment]
Side note: for the PR description if you write `Fixes #4314` it will automatically close the issue when this PR is merged.

**Comment by JudyWeng** [comment]
I just fixed this file based on the comments. Please let me know if there is anything else I should fix.

**Comment by lilleyse** [comment]
Those changes look good.

**Pull request merged**